crates/ra_ide/src/completion/complete_keyword.rs

```diff
@@ -52,7 +53,28 @@ fn keyword(ctx: &CompletionContext, kw: &str, snippet: &str) -> CompletionItem {
     .build()
 }
 
+fn add_top_level_keywords(acc: &mut Completions, ctx: &CompletionContext) {
+    if let Some(token) = previous_non_triva_element(&ctx.token).and_then(|it| it.into_token()) {
+        if token.kind() == UNSAFE_KW {
+            acc.add(keyword(ctx, "impl", "impl $0 {}"));
+            acc.add(keyword(ctx, "trait", "trait $0 {}"));
+            acc.add(keyword(ctx, "fn", "fn $0() {}"));
+            return;
+        }
+    }
+    acc.add(keyword(ctx, "impl", "impl $0 {}"));
+    acc.add(keyword(ctx, "enum", "enum $0 {}"));
+    acc.add(keyword(ctx, "struct", "struct $0 {}"));
+    acc.add(keyword(ctx, "trait", "trait $0 {}"));
+    acc.add(keyword(ctx, "fn", "fn $0() {}"));
+    acc.add(keyword(ctx, "unsafe", "unsafe "));
```

Comment:
I haven't checked, but are we missing `mod`, `const`, `static` and `union`?

Reply:
Yes, you are right I will add them and update the PR. Thanks!

Reply:
Might be worth looking at https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/ra_parser/src/grammar/items.rs#L79 and https://github.com/rust-analyzer/rust-analyzer/blob/master/crates/ra_parser/src/grammar/items.rs#L230 for an exhaustive list.

Reply:
CC #4737